android/README.md

```diff
@@ -33,80 +33,72 @@ dependencies {
-the above will suffice. If you would like to publish a third-party react-native module
-which Jitsi Meet SDK for Android depends on (and is not publicly available in Maven
-repositories) continue below.
+    $ cp -r ../node_modules/jsc-android/dist/org /tmp/repo/
 
-6. Create the release assembly for _each_ third-party react-native module that you
-   need, replacing it's name in the example below.
+Third-party React Native _modules_, which Jitsi Meet SDK for Android depends on, are download by NPM in source code form. These need to be assembled into Maven artifacts, and then published to your local Maven repository. The SDK project facilitates this. 
 
-   ```bash
-   ./gradlew :react-native-webrtc:assembleRelease
-   ```
+To prepare, Configure the Maven repositories in which you are going to publish the SDK artifacts/binaries. In `android/sdk/build.gradle` as well as in `android/build.gradle` modify the lines that contain:
 
-7. Configure the Maven repositories in which you are going to publish the
-   artifacts/binaries during step 8.
+    "file:${rootProject.projectDir}/../../../jitsi/jitsi-maven-repository/releases"
 
-   In the file `android/build.gradle` (note that this is a different file than the file
-   that was modified in step 3) modify the line that contains
-   `"file:${rootProject.projectDir}/../../../jitsi/jitsi-maven-repository/releases"`
+Change this value (which represents the Maven repository location used internally by the Jitsi Developers) to the location of the repository that you'd like to use:
 
-   Change this value (which represents the Maven repository location used internally
-   by the Jitsi Developers) to the location of the repository that you'd like to use.
-   You can use the same repository as the one you configured in step 3 if you want.
+    "file:/tmp/repo"
 
-8. Publish the Maven artifact/binary of _each_ third-party react-native module that
-   you need, replacing it's name in the example below. For example, to publish
-   react-native-webrtc:
+Make sure to do this in both files! Each file should require one line to be changed.
 
-   ```bash
-   ./gradlew :react-native-webrtc:publish
-   ```
+To create the release assembly for any _specific_ third-party React Native module that you need, you can execture the following commands, replace the module name in the examples below.
 
-   Note that there should not be a need to explicitly add these dependencies in
-   _your_ project, as they will be pulled in as transitive dependencies of
-   `jitsi-meet-sdk`.
+    $ ./gradlew :react-native-webrtc:assembleRelease 
+    $ ./gradlew :react-native-webrtc:publish
+
+You build and publish the SDK itself in the same way:
+
+    $ ./gradlew :sdk:assembleRelease
+    $ ./gradlew :sdk:publish
+
+Alternatively, you can assemble and publish _all_ subprojects, which include the react-native modules, but also the SDK itself, with a single command:
+
+    $ ./gradlew assembleRelease publish
```

Comment:
Have you checked that this command now publishes jsc as well?

Reply:
_that_ command does not. jsc is downloaded by NPM in binary form. It needs to be copied to the Maven repository manually, just like you need to do with React Native (the core thingy, not the modules). I added instructions a couple lines higher: it's a simple recursive copy.

Reply:
Hum, But I thought we had some gradle magic to publish those artifacts as packages... Maybe I'm misremembering. I'll take a look.

Reply:
From what I can tell, there are Gradle modules in the project that also holds the SDK module, that directly link to the source code of various react-native dependencies in `../node_modules`. This is what appears to link the Gradle module to the node_modules source: https://github.com/jitsi/jitsi-meet/blob/56100d0d5c99826495fbab5eec8e4010a71ac947/android/settings.gradle#L4-L25 This is not done for React Native itself, presumably because the corresponding node_module already contains the binary artifact (there's no need to build this ourselves). The same is true for jsc: there's already a binary artifact (and my half-baked attempt at trying to build one in the same manner as those React modules failed).

Reply:
So, to build a Maven repository, one of two things need to happen:

1. copy the binary artifact from `../node_modules/` to the maven repo
2. build the binary artifact from the source in `../node_modules` and publish it to the maven repo

The latter is needed only when the node_modules do not include a binary. For these modules, there's the Gradle magic to make those node_modules also gradle modules, which then can easily be built en published through the SDK project.